### ISSUE TITLE
dotnet publish -c whatever

### ISSUE THREAD

**Issue by seesharper** [dotnet/cli#9072]
## Steps to reproduce

```
dotnet new console
dotnet publish -c whatever
```
## Expected  behavior

Some kind of error message regarding invalid configuration

## Actual behavior

Published with the `whatever` configuration whatever that might be, Release or Debug ?

```
  Restore completed in 36.56 ms for /Users/bernhardrichter/GitHub/ConsoleApp/ConsoleApp.csproj.
  ConsoleApp -> /Users/bernhardrichter/GitHub/ConsoleApp/bin/whatever/netcoreapp2.1/ConsoleApp.dll
  ConsoleApp -> /Users/bernhardrichter/GitHub/ConsoleApp/bin/whatever/netcoreapp2.1/publish/
```
## Environment data
`dotnet --info` output:

```
.NET Command Line Tools (2.1.300-preview1-008174)

Product Information:
 Version:            2.1.300-preview1-008174
 Commit SHA-1 hash:  b8df89a54f

Runtime Environment:
 OS Name:     Mac OS X
 OS Version:  10.13
 OS Platform: Darwin
 RID:         osx.10.12-x64
 Base Path:   /usr/local/share/dotnet/sdk/2.1.300-preview1-008174/

Microsoft .NET Core Shared Framework Host

  Version  : 2.1.0-preview1-26216-03
  Build    : f2c3216183d20416568a4bbf5bb7d153e826f153
```

**Comment by livarcocc** [comment]
I believe this is by design. There is no fixed set of possible values for Configuration, as you can define your own in your project files and build.

**Comment by andmos** [comment]
So let's say we run `dotnet publish -c whatever`, the `bin/whatever/netcoreapp2.1/publish` DLL's will be Debug, since there is no `whatever` configuration? I'm also curious about uppercase in `Release` vs `release`, any difference there?

**Comment by dasMulli** [comment]
msbuild string comparisons are case insensitive by default so the defaults should apply to Debug/debug/DeBuG in any case..

cc @dsplaisted maybe.. since VS 15.3+ introduced the `<Configurations>` property, I was wondering if it would be interesting to issue a warning if a project is built/published with a `$(Configuration)` that is not listed in `$(Configurations)`. Not sure what the tooling support for `$(Configurations)` currently is in VS/Mac. (Rider implemented it for 2018.1)

**Comment by andmos** [comment]
Thanks for the answer. We stumbled upon this after a typo, as you might guess 😆
